src/status_im/ui/screens/profile/contact/views.cljs

```diff
@@ -204,20 +159,24 @@
 
 (defn profile
   []
-  (let [{:keys [public-key name ens-verified] :as contact} @(re-frame/subscribe
-                                                             [:contacts/current-contact])
-        muted?                                             @(re-frame/subscribe [:chats/muted
-                                                                                 public-key])
-        pinned-messages                                    @(re-frame/subscribe [:chats/pinned
-                                                                                 public-key])
-        [first-name second-name]                           (multiaccounts/contact-two-names contact true)
-        on-share                                           #(re-frame/dispatch
-                                                             [:show-popover
-                                                              (merge
-                                                               {:view    :share-chat-key
-                                                                :address public-key}
-                                                               (when (and ens-verified name)
-                                                                 {:ens-name name}))])]
+  (let [{:keys [public-key
+                name
+                ens-verified
+                compressed-key]
+         :as   contact}
+        @(re-frame/subscribe
+          [:contacts/current-contact])
+        muted? @(re-frame/subscribe [:chats/muted
+                                     public-key])
+        [first-name second-name] (multiaccounts/contact-two-names contact true)
+        on-share #(re-frame/dispatch
+                   [:show-popover
+                    (merge
+                     {:view    :share-chat-key
+                      :address (or compressed-key
```

Comment:
Quick question: I see in the specs nodes SHOULD provide compression/decompression of public keys. Are we planning to eventually change the spec to force all nodes to have this capability?

Reply:
I don't think so, being a `MUST` is not necessary, since it's client dependent, for example if we started from compressed key from the get go, we would not need compression/decompression, and I can imagine soon enough we won't really need that, since the best way is to have status-go accept both format, and keep the conversion there, so it's transparent to clients and they can use them interchangeably.

Eventually, we could fade out the usage of the old key format and just index by the new key-id I suppose.

Reply:
>keep the conversion there, so it's transparent to clients and they can use them interchangeably.

Understood, it does make sense :)